syntax = "proto3";

package flowpipe.stages.s3.source.v1;

// Configuration for S3Source stage
message S3SourceConfig {
  // S3 bucket name
  string bucket = 1;

  // Object key to read
  string key = 2;

  // AWS region (e.g., us-east-1)
  string region = 3;

  // Optional endpoint override (for S3-compatible storage)
  string endpoint = 4;

  // Use path-style addressing instead of virtual-hosted style
  bool use_path_style = 5;

  // Optional AWS access key ID
  string access_key_id = 6;

  // Optional AWS secret access key
  string secret_access_key = 7;

  // Optional AWS session token
  string session_token = 8;
}
